pkg/admissioncontroller/webhooks/validate_resource_size.go

```diff
@@ -0,0 +1,171 @@
+// limitations under the License.
+
+package webhooks
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-logr/logr"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	apisconfig "github.com/gardener/gardener/pkg/admissioncontroller/apis/config"
+	confighelper "github.com/gardener/gardener/pkg/admissioncontroller/apis/config/helper"
+	"github.com/gardener/gardener/pkg/admissioncontroller/metrics"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/utils"
+)
+
+const resourceSizeValidatorName = "resource_size_validator"
+
+var _ admission.Handler = &ObjectSizeHandler{}
+
+// ObjectSizeHandler is a webhook handler for validating that the resource size of a request doesn't exceed the
+// configured limits.
+type ObjectSizeHandler struct {
+	// Config is the configuration for this webhook handler.
+	Config *apisconfig.ResourceAdmissionConfiguration
+
+	logger  logr.Logger
+	decoder *admission.Decoder
+}
+
+// InjectLogger injects a logger into the handler.
+func (h *ObjectSizeHandler) InjectLogger(l logr.Logger) error {
+	h.logger = l.WithName(resourceSizeValidatorName)
+	return nil
+}
+
+// InjectDecoder injects a decoder capable of decoding objects included in admission requests.
+func (h *ObjectSizeHandler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+// Handle implements the webhook handler for resource size validation.
+func (h *ObjectSizeHandler) Handle(_ context.Context, request admission.Request) admission.Response {
+	requestID, err := utils.GenerateRandomString(8)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	requestLogger := h.logger.WithValues(logger.IDFieldName, requestID, "user", request.UserInfo.Username,
+		"resource", request.Resource, "name", request.Name)
+	if request.Namespace != "" {
+		requestLogger = requestLogger.WithValues("namespace", request.Namespace)
+	}
+
+	response := h.admitRequestSize(request, requestLogger)
+	if !response.Allowed {
+		metrics.RejectedResources.WithLabelValues(
+			fmt.Sprint(request.Operation),
+			request.Kind.Kind,
+			request.Namespace,
+			metrics.ReasonSizeExceeded,
+		).Inc()
+	}
+	return response
+}
+
+// admitRequestSize only allows the request if the object in the request does not exceed a configured limit.
+func (h *ObjectSizeHandler) admitRequestSize(request admission.Request, requestLogger logr.Logger) admission.Response {
+	if request.SubResource != "" {
+		// TODO: why do we have `shoot/status` in the dev-setup webhhok then?
+		// https://github.com/gardener/gardener/blob/1202ab60796a3ef98ca81da0f80d1a885fc782e4/hack/local-development/dev-setup-register-gardener#L324
```

Comment:
Good point. I think this was just a left-over for testing. Feel free to remove it from the dev-setup.